ExampleMod/Projectiles/ExampleSolarEruptionExplosion.cs

```diff
@@ -0,0 +1,109 @@
+﻿using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using Terraria;
+using Terraria.Graphics.Shaders;
+using Terraria.ID;
+using Terraria.ModLoader;
+
+namespace ExampleMod.Projectiles
+{
+	public class ExampleSolarEruptionExplosion : ModProjectile
+	{
+		public override void SetStaticDefaults()
+		{
+			DisplayName.SetDefault("Example Solar Eruption Explosion");
+			Main.projFrames[projectile.type] = 5;
+		}
+
+		public override void SetDefaults()
+		{
+			projectile.width = 52;
+			projectile.height = 52;
+			projectile.friendly = true;
+			projectile.alpha = 255;
+			projectile.ignoreWater = true;
+			projectile.timeLeft = 60;
+			projectile.tileCollide = false;
+			projectile.penetrate = -1;
+			projectile.usesLocalNPCImmunity = true;

```

Comment:
I feel like stuff such as this should have comments explaining what they actually do.